### PR TITLE
Add initial macOS content

### PR DESCRIPTION
**Pull request by redhatrises**
- Helps NIST for building macOS SCAP content and removes some barriers of SCAP creation for them

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -70,6 +70,7 @@ option(SSG_PRODUCT_FEDORA "If enabled, the Fedora SCAP content will be built" ${
 option(SSG_PRODUCT_FIREFOX "If enabled, the Firefox SCAP content will be built" ${SSG_PRODUCT_DEFAULT})
 option(SSG_PRODUCT_FUSE6 "If enabled, the JBoss Fuse6 SCAP content will be built" ${SSG_PRODUCT_DEFAULT})
 option(SSG_PRODUCT_JRE "If enabled, the JRE SCAP content will be built" ${SSG_PRODUCT_DEFAULT})
+option(SSG_PRODUCT_MACOS1015 "If enabled, the JRE SCAP content will be built" ${SSG_PRODUCT_DEFAULT})
```
Comment (shawndwells) [inline CMakeLists.txt]:
```suggestion
option(SSG_PRODUCT_MACOS1015 "If enabled, the Mac OSX SCAP content will be built" ${SSG_PRODUCT_DEFAULT})
```

File: apple_os/auditing/configure_auditing/audit_failure_halt/bash/shared.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# platform = Apple macOS 10.15
+# reboot = false
+# strategy = enable
+# complexity = low
+# disruption = low
+
+/usr/bin/sed -i.bak '/^policy/ s/$/,ahlt/' /etc/security/audit_control; sudo /usr/sbin/audit -s
```
Comment (shawndwells) [inline apple_os/auditing/configure_auditing/audit_failure_halt/bash/shared.sh]: Was ``ahlt`` meant to be ``halt``?

Comment (redhatrises) [reply apple_os/auditing/configure_auditing/audit_failure_halt/bash/shared.sh]: It's `ahlt`. I did a double take on it as well.

File: CMakeLists.txt (hunk 1 above)
Comment (redhatrises) [reply CMakeLists.txt]: Updated to `Apple macOS 10.15`

**Comment by redhatrises** [comment]
> Potential typos to clarify.
> 
> Also, the content is dropped into `apple_os`. Should that be `macos` to keep things consistent?

There's more operating systems in the apple portfolio, so I think keeping it `apple_os` makes the most sense.

> Since many of these overlap with existing checks, should they be put into `shared`? Would require updating the entire shared build system so perceiving that as a future to-do... but wanted to talk through expectations.

Not exactly sure what you are talking about, but no, let's not muddy the waters between apple and linux content.

> 
> Nice start!

**Comment by shawndwells** [comment]
> > Potential typos to clarify.
> > Also, the content is dropped into `apple_os`. Should that be `macos` to keep things consistent?
> 
> There's more operating systems in the apple portfolio, so I think keeping it `apple_os` makes the most sense.

So, ``apple_os/macOS`` then? To potentially prepare for ``apple_os/iOS``, ``apple_os/macOS`` etc?

> 
> > Since many of these overlap with existing checks, should they be put into `shared`? Would require updating the entire shared build system so perceiving that as a future to-do... but wanted to talk through expectations.
> 
> Not exactly sure what you are talking about, but no, let's not muddy the waters between apple and linux content.
> 
> > Nice start!

**Comment by redhatrises** [comment]
> > > Potential typos to clarify.
> > > Also, the content is dropped into `apple_os`. Should that be `macos` to keep things consistent?
> > 
> > 
> > There's more operating systems in the apple portfolio, so I think keeping it `apple_os` makes the most sense.
> 
> So, `apple_os/macOS` then? To potentially prepare for `apple_os/iOS`, `apple_os/macOS` etc?

Preferably not. Much like how the `linux_os`, the `apple_os` should work the same way.

> > > Since many of these overlap with existing checks, should they be put into `shared`? Would require updating the entire shared build system so perceiving that as a future to-do... but wanted to talk through expectations.
> > 
> > 
> > Not exactly sure what you are talking about, but no, let's not muddy the waters between apple and linux content.
> > > Nice start!

**Comment by redhatrises** [comment]
Also Apple products != Linux technologies. It would be good to keep these technologies separate.

**Comment by evgenyz** [comment]
Mac OS X (starting from 10.5, except 10.7) is literally Unix, though. On the other hand iOS is not Unix at all, it's Unix-like. It shares kernel and bits with OS X, but it was never acknowledged as a Unix.

**Comment by redhatrises** [comment]
Regardless most people are going to logically think about content in terms of Mac, Windows, Linux, etc., and we should keep that and prevent confusion and unnecessary complexity.

File: macos1015/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+ssg_build_html_table_by_ref(${PRODUCT} "nist")
+ssg_build_html_table_by_ref(${PRODUCT} "cui")
+ssg_build_html_table_by_ref(${PRODUCT} "cis")
+ssg_build_html_table_by_ref(${PRODUCT} "pcidss")
+ssg_build_html_table_by_ref(${PRODUCT} "anssi")
+
+ssg_build_html_nistrefs_table(${PRODUCT} "standard")
+#ssg_build_html_nistrefs_table(${PRODUCT} "ospp")
+ssg_build_html_nistrefs_table(${PRODUCT} "stig")
```
Comment (shawndwells) [inline macos1015/CMakeLists.txt]: Do these profiles exist? Looks like only a ``moderate`` profile was included.

Comment (redhatrises) [reply macos1015/CMakeLists.txt]: Good call... fixed

File: apple_os/benchmark.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+status: draft
+
+description: |
+    This guide presents a catalog of security-relevant
+    configuration settings for {{{ full_name }}}. It is a rendering of
+    content structured in the eXtensible Configuration Checklist Description Format (XCCDF)
+    in order to support security automation.  The SCAP content is
+    is available in the <tt>scap-security-guide</tt> package which is developed at
+    {{{ weblink(link="https://www.open-scap.org/security-policies/scap-security-guide") }}}.
+    <br/><br/>
```
[the remark this answers is not in the document]
Comment (redhatrises) [reply apple_os/benchmark.yml]: It does not. Fixed

File: apple_os/auditing/service_auditd_enabled/rule.yml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+description: |-
+    The <tt>audit</tt> service is an essential userspace component of
+    the auditing system, as it is responsible for writing audit records to
+    disk.
+    {{{ describe_service_enable(service="auditd") }}}
+
+rationale: |-
+    Without establishing what type of events occurred, when they
+    occurred, and by whom it would be difficult to establish, correlate,
```
[the remark this answers is not in the document]
Comment (redhatrises) [reply apple_os/auditing/service_auditd_enabled/rule.yml]: fixed

File: apple_os/auditing/group.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+description: |-
+    The Basic Security Module (BSM) security audit API and file format is Apple's
+    auditing system. The audit() function submits a record to the kernel for inclusion in the
+    global audit trail. The record must already be in BSM format. To protect
+    the integrity of the audit trail, this system call must be made with sufficient
+    privileges. Libbsm can be used to create and manipulate BSM data. Length
+    is the length in bytes of the BSM record and record points to the data.
+    The audit service provides substantial capabilities for recording system
+    activities. Government networks often have substantial auditing
```
[the remark this answers is not in the document]
Comment (redhatrises) [reply apple_os/auditing/group.yml]: Fixed

**Comment by shawndwells** [comment]
ack pending jenkins

**Pull request merged**